src/goto-symex/goto_symex_state.cpp

```diff
@@ -342,6 +342,10 @@ void goto_symex_statet::assignment(
   assert_l2_renaming(lhs);
   assert_l2_renaming(rhs);
 
+  // see #305 on GitHub for a simple example and possible discussion
+  if(is_shared && lhs.type().id() == ID_pointer)
+    throw "pointer handling for concurrency is unsound";
```

Comment:
Note: that should be turned into a user facing exception (not an invariant) during the error handling clean up.